### PR TITLE
IIT B English to Hindi machine translation dataset

### PR DESCRIPTION
**Pull request by spatil6**
Adding IIT Bombay English-Hindi Corpus dataset
more info : http://www.cfilt.iitb.ac.in/iitb_parallel/

### CODE REVIEW

File: datasets/iitb_en_hi/iitb_en_hi.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "input_filepath_english": os.path.join(data_dir["dev-test"], "dev_test", "dev.en"),
+                    "input_filepath_hindi": os.path.join(data_dir["dev-test"], "dev_test", "dev.hi"),
+                },
+            ),
+        ]
+
+    def _generate_examples(self, input_filepath_english, input_filepath_hindi):
+        with open(input_filepath_english) as en_texts, open(input_filepath_hindi) as hi_texts:
```
Comment (sumanthd17) [inline datasets/iitb_en_hi/iitb_en_hi.py:94]: use utf-8 encoding
i.e with open(filepath, encoding='utf-8')

**Review by sumanthd17 — COMMENTED** [review]
Also make sure to run `make style` and `flake8 datasets`

**Comment by abhishekkrthakur** [comment]
Please note that this dataset is actually behind a form that one needs to fill. However, the link is direct. I'm not sure what should the approach be in this case.

File: datasets/iitb_en_hi/README.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
+---
+YAML tags:
```
Comment (lhoestq) [inline datasets/iitb_en_hi/README.md:2]:
```suggestion
```

File: datasets/iitb_en_hi/iitb_en_hi.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        )
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "english_text": datasets.Value("string"),
+                    "hindi_text": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/iitb_en_hi/iitb_en_hi.py:58]: can you use the `Translation` feature type ?
you can see an example of usage in [flores](https://github.com/huggingface/datasets/blob/master/datasets/flores/flores.py)

**Review by lhoestq — COMMENTED** [review]
Looks very cool !
thanks for adding this one

I left a comment about the feature type that needs to be `Translation`. Could you do that change please ?

**Comment by lhoestq** [comment]
also pinging @thomwolf 
The dataset webpage returns a form when trying to download the dataset (form here : http://www.cfilt.iitb.ac.in/iitb_parallel/dataset.html).
However the url we get with the form can be used for the dataset script.
Should we ask the authors or use the urls this way ?

**Comment by spatil6** [comment]
> also pinging @thomwolf
> The dataset webpage returns a form when trying to download the dataset (form here : http://www.cfilt.iitb.ac.in/iitb_parallel/dataset.html).
> However the url we get with the form can be used for the dataset script.
> Should we ask the authors or use the urls this way ?

I had discussion on this with @thomwolf . We have already sent email to author of this dataset.

File: datasets/iitb_en_hi/iitb_en_hi.py (hunk 3 above)
Comment (spatil6) [reply datasets/iitb_en_hi/iitb_en_hi.py:58]: sure

File: datasets/iitb_en_hi/iitb_en_hi.py (hunk 1 above)
Comment (spatil6) [reply datasets/iitb_en_hi/iitb_en_hi.py:94]: sure

**Comment by lhoestq** [comment]
Hi @spatil6 !
Any news from the authors ?

**Comment by spatil6** [comment]
IIT B folks will add this dataset to repo.

**Pull request closed without merging**